core/types/pow/config.go

```diff
@@ -99,14 +101,18 @@ func (this *PowConfig) Check() error {
 			return errors.New("pow config error, mainHeight set repeat!")
 		}
 		heightArr[p.MainHeight] = 1
-		if p.CuckarooPercent < 0 || p.Blake2bDPercent < 0 ||
-			p.CuckatooPercent < 0 || p.CuckaroomPercent < 0 ||
-			p.QitmeerKeccak256Percent < 0 ||
-			p.X16rv3Percent < 0 || p.X8r16Percent < 0 {
-			return errors.New("pow config error, all percent must greater than or equal to 0!")
+		typ := reflect.TypeOf(p)
+		val := reflect.ValueOf(p) //reflect.Type
+		for i := 0; i < val.NumField(); i++ {
+			field := val.Field(i)
+			tagVal := typ.Field(i).Tag.Get("json")
+			if strings.Contains(tagVal, "_percent") {
+				if field.Interface().(int) < 0 {
+					return errors.New("pow config error, all percent must greater than or equal to 0!")
+				}
+				allPercent += field.Interface().(int)
+			}
```

Comment:
1.) more error-checking required on the case that when field contains no `json:xxx_percent`. 
2.) the `MainHeight` field should not be a member of type `percent` , because it means what's percent is at a specified main-height.